sbi/inference/base.py

```diff
@@ -453,40 +407,64 @@ def _summarize(
     def summary(self):
         return self._summary
 
-    def _check_proposal(self, proposal):
-        """
-        Check for validity of the provided proposal distribution.
-
-        If the proposal is a `NeuralPosterior`, we check if the default_x is set and
-        if it matches the `_x_o_training_focused_on`.
-
-        If the proposal is **not** a `NeuralPosterior`, we warn. This is especially
-        important if the user passed the prior as proposal. Consider e.g.:
-        ```
-        posterior1 = infer(num_simulations=200, proposal=prior)
-        posterior2 = infer(num_simulations=200, proposal=prior)
-        ```
-        This will trigger atomic loss in the second line, which is not wanted.
-        """
-        if proposal is not None:
-            if isinstance(proposal, NeuralPosterior):
-                if proposal.default_x is None:
-                    raise ValueError(
-                        "`proposal.default_x` is None, i.e. there is no "
-                        "x_o for training. Set it with "
-                        "`posterior.set_default_x(x_o)`."
-                    )
-            else:
-                warn(
-                    "The proposal you passed is not a `NeuralPosterior` object. If you "
-                    "are an expert user and did so for research purposes, this is fine."
-                    " If not, and you only wanted to do single round inference with"
-                    " `proposal=prior`, please instead set `proposal=None`, which"
-                    " automatically uses the prior as proposal."
-                )
-        elif self._round > 0:
+
+def simulate_for_sbi(
```

Comment:
I think I'd rather keep it were it is right now:
1) I'd consider inference to be: simulate, train, build posterior (so simulation is part of inference)
2) It might rely on functionality from `sbi.inference`. For now, it only does some checks to see if the `proposal` has a `default_x`, but even these lead to circular imports if we just move it to `simulators/simutils.py`. In the future, if we end up doing a bit more with the simulator (e.g. train a classifier to reject `NaN` simulations), there might be more overlaps.